torch/_dynamo/variables/misc.py

```diff
@@ -649,26 +649,34 @@ def visit(node):
 
         VariableTracker.visit(visit, (args, kwargs))
 
+        from torch._functorch.autograd_function import (
+            autograd_function_forward_rewritten,
+        )
+        from torch.autograd.function import _is_setup_context_defined
+
+        forward_fn = self.fn_cls.forward
+
+        is_setup_ctx_defined = _is_setup_context_defined(self.fn_cls.setup_context)
+        if is_setup_ctx_defined:
+            # If setup_context is defined, we generate a new forward function which includes
+            # the original forward and setup_context function, and trace the new forward function.
+            forward_fn = autograd_function_forward_rewritten(
+                self.fn_cls.forward, self.fn_cls.setup_context
+            )
+        else:
+            if torch._C._are_functorch_transforms_active():
+                raise RuntimeError(
```

Comment:
Not sure if this should be ```RuntimeError```, I copied this from [here](https://github.com/pytorch/pytorch/blob/7a554e96b4ca1fc8c45a49604c0151a784776296/torch/autograd/function.py#L578) to make eager and compile's behavior consistent. This is tested by:
https://github.com/pytorch/pytorch/blob/7a554e96b4ca1fc8c45a49604c0151a784776296/test/functorch/test_eager_transforms.py#L3563